runtime/src/sdk.rs

```diff
@@ -5,74 +5,135 @@ use std::str::FromStr;
 
 use crate::*;
 
+/// Create a new context
 #[no_mangle]
-pub unsafe extern "C" fn extism_plugin_register(
+pub unsafe extern "C" fn extism_context_new() -> *mut Context {
+    trace!("Creating new Context");
+    Box::into_raw(Box::new(Context::new()))
+}
+
+/// Free a context
+#[no_mangle]
+pub unsafe extern "C" fn extism_context_free(ctx: *mut Context) {
+    trace!("Freeing context");
+    if ctx.is_null() {
+        return;
+    }
+    drop(Box::from_raw(ctx))
```

Comment:
Should we iterate the plugins here and free them? It would prevent the progammers from needing to track and clean up the plugins they make in it's scope.

Reply:
I believe this does clean up all the plugins automatically since they’re just references to values stored in the `plugins` field. The only reason `extism_plugin_free` exists is if you specifically want the context to outlive the plugin. But now that the context stuff exists we can de-emphasize the need to free plugins.

Reply:
I see, makes sense! I need to re-read about rust lifetimes.